### PR TITLE
Various Build fixes

### PR DESCRIPTION
**Pull request by ycheng627**
Various independent changes that changes the build system and config. Will rebase-and-merge after approved.

See individual CL descriptions.

### CODE REVIEW

File: diagnostics-app/src/app/layout/header/header.component.ts [hunk 1]
```diff
@@ -18,6 +19,7 @@ import { ThemeService } from 'src/app/core/services/theme.service';
 })
 export class HeaderComponent {
   @Output() toggleDrawer = new EventEmitter<void>();
+  app_name = APP_NAME;
```
Comment (andy920262) [inline diagnostics-app/src/app/layout/header/header.component.ts:22]: `readonly`

File: diagnostics-extension/public/manifest.json [hunk 2]
```diff
@@ -3,23 +3,17 @@
   "version": "1.9.0",
   "description": "The companion extension for the ChromeOS Diagnostics App.",
```
[the remark this answers is not in the document]
Comment (ycheng627) [reply diagnostics-extension/public/manifest.json]: Done.

I've decided to use `environment.ts` and `manifest.json` as the default values for production, and additionally created a `environment.dev.ts` and `manifest.dev.json` file.

In order to use the replacement module (before compiling), we must already have a environment.ts file present. Thus we can't use `environment.prod.ts` and `environment.dev.ts`.

File: diagnostics-app/src/app/layout/header/header.component.ts (hunk 1 above)
Comment (ycheng627) [reply diagnostics-app/src/app/layout/header/header.component.ts]: Done.

**Review by andy920262 — APPROVED** [review]

File: diagnostics-app/src/environments/environment.pwa.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
```
Comment (BaiiYuan) [inline diagnostics-app/src/environments/environment.pwa.ts]: nit: 2023

**Review by BaiiYuan — APPROVED** [review]
LGTM

File: diagnostics-app/src/environments/environment.pwa.ts (hunk 3 above)
Comment (ycheng627) [reply diagnostics-app/src/environments/environment.pwa.ts]: Done.

**Comment by ycheng627** [comment]
Hi Reviewers,

Added one more small change in the extension build CL.

Add some jest config to make sure that unit tests uses test mode, as to always return fake data.

PTAL!

**Comment by BaiiYuan** [comment]
Still LGTM.

One question: Do we need to add the license header to all `js` files?  (we can revisit them later)

**Pull request merged**